core/src/test/scala/org/broadinstitute/dsde/rawls/entities/local/LocalEntityProviderSpec.scala

```diff
@@ -82,6 +82,18 @@ class LocalEntityProviderSpec extends AnyWordSpecLike with Matchers with TestDri
         Map(sampleSet2.name -> Seq(SubmissionValidationValue(Some(AttributeValueEmptyList), None, intArrayNameWithWfName)))
     }
 
+    "resolve empty lists into empty Array in nested WDL Struct" in withConfigData {
+      val context = workspace
```

Comment:
Nit: you could refactor these test points to share the common code. The inputs and expected outputs vary, but everything else is the same.

Reply:
Since this is test code, I am inclined to leave it like this as in my opinion time involved in refactoring outweighs the benefits

Reply:
Having said that, if you do feel strongly, I can refactor them :)

Reply:
I was able to find [this](https://github.com/broadinstitute/rawls/blob/7bc2274032277fd9605495fbeb4c7a97daefc370/core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/SubmissionApiServiceSpec.scala#L837-L872) pattern. Is this what you had in mind?

Reply:
Yes, but I was wondering if there were language constructs/libraries to make this easier. For example, the ddt library in Python:

````python
    @data(
        ['Alewife', 'Alewife', [['Red Line']]],
        ['Shawmut', 'Quincy Adams', [['Red Line']]],
        ['Downtown Crossing', 'Alewife', [['Red Line']]],
        ['Downtown Crossing', 'Haymarket', [['Orange Line']]],
        ['Ashmont', 'Milton', [['Mattapan Trolley']]],
        ['Haymarket', 'Prudential', [['Green Line E']]],
        ['Park Street', 'North Station', [['Green Line C'], ['Green Line E']]],
        ['Ruggles', 'Wonderland', [['Orange Line', 'Blue Line']]],
        ['South Station', 'State', [['Red Line', 'Orange Line']]],
        ['Copley', 'Chinatown', [['Green Line C', 'Orange Line'],
                                 ['Green Line E', 'Orange Line']]],
        ['Fenway', 'North Station', [['Green Line D', 'Green Line C'],
                                     ['Green Line D', 'Green Line E']]],
        ['Mattapan', 'Roxbury Crossing', [['Mattapan Trolley', 'Red Line', 'Orange Line']]],
        ['Broadway', 'Aquarium', [['Red Line', 'Orange Line', 'Blue Line'],
                                  ['Red Line', 'Green Line C', 'Blue Line'],
                                  ['Red Line', 'Green Line D', 'Blue Line'],
                                  ['Red Line', 'Green Line E', 'Blue Line']]],
        ['Fenway', 'Assembly', [['Green Line D', 'Green Line C', 'Orange Line'],
                                ['Green Line D', 'Green Line E', 'Orange Line'],
                                ['Green Line D', 'Blue Line', 'Orange Line']]]
    )
    @unpack
    def test_get_route(self, source, destination, possible_routes):
        """
        Tests the get_route method (Question 3). If multiple routes are possible,
        makes sure that the route returned is in the list of possible routes.
        """
        route = self.subway.get_route(source, destination)
        self.assertIn(
            route, possible_routes,
            'Route incorrect for stops {source} -> {destination}. Expected one of {possible_routes}, but got {route}.'.format(
                source=source, destination=destination, possible_routes=possible_routes, route=route
            )
        )